messages/src/lib.rs

```diff
@@ -108,106 +129,61 @@ impl PortMask {
     pub fn to_indices(&self) -> impl Iterator<Item = u8> + '_ {
-        (0..Self::NUM_PORTS).filter(|i| self.is_set(*i).unwrap())
+        (0..Self::MAX_INDEX).filter(|i| self.is_set(*i).unwrap())
     }
 
     /// A convenience function to return a port bitmask identifying a single
     /// port by index.
     pub const fn single(index: u8) -> Result<Self, Error> {
-        if index >= Self::NUM_PORTS {
+        if index >= Self::MAX_INDEX {
             Err(Error::InvalidPort(index))
         } else {
             Ok(Self(1 << index))
         }
     }
 }
 
-impl From<u32> for PortMask {
-    fn from(x: u32) -> Self {
-        Self(x)
-    }
-}
-
-impl From<PortMask> for u32 {
-    fn from(m: PortMask) -> Self {
-        m.0
-    }
-}
-
-/// Identifier for an FPGA on a Sidecar.
-#[derive(Clone, Copy, Debug, Deserialize, Eq, PartialEq, Serialize, SerializedSize)]
-pub struct Fpga(u8);
-
-impl Fpga {
-    pub const LEFT: Self = Self(0);
-    pub const RIGHT: Self = Self(1);
-}
-
-impl TryFrom<u8> for Fpga {
-    type Error = Error;
-    fn try_from(x: u8) -> Result<Self, Self::Error> {
-        let maybe_self = Self(x);
-        match maybe_self {
-            Self::LEFT | Self::RIGHT => Ok(maybe_self),
-            _ => Err(Error::InvalidFpga(x)),
-        }
-    }
-}
-
-/// A unique identifier for a set of transceiver modules on a Sidecar.
-///
-/// Modules are identified by a combination of the FPGA and a bitmask of ports
-/// on that FPGA.
-#[derive(Clone, Copy, Debug, Deserialize, PartialEq, Serialize, SerializedSize)]
+/// Identifier for a set of transceiver modules accessed through a single FPGA.
+#[derive(Clone, Copy, Debug, Default, Deserialize, PartialEq, Serialize, SerializedSize)]
 pub struct ModuleId {
-    pub fpga: Fpga,
+    pub fpga_id: u8,
     pub ports: PortMask,
 }
 
-impl core::fmt::UpperHex for ModuleId {
-    fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
-        write!(f, "0x{:X}:0x{:X}", self.fpga.0, self.ports)
-    }
-}
-
-// Map a logical port to the FPGA and per-FPGA port index.
-pub fn port_to_ids(port: u8) -> Result<(Fpga, u8), Error> {
-    // There are 16 ports per FPGA, with a stride of 8. That is:
-    //
-    // Logical port     FPGA    On-FPGA port
-    // ------------     ----    ------------
-    // 0-8              0       0-8
-    // 8-16             1       0-8
-    // 16-24            0       9-16
-    // 24-32            1       9-16.
-    //
-    // The below basically implements poor-man's 2D array indexing from a single
-    // flattened index, assuming row-major ordering.
-    const STRIDE: u8 = 8;
-    const PORTS_PER_FPGA: u8 = 16;
-    const NUM_FPGAS: u8 = 2;
-    let fpga_id = (port / STRIDE) % NUM_FPGAS;
-    let port_index = ((port / PORTS_PER_FPGA) * STRIDE) + (port % STRIDE);
-    Fpga::try_from(fpga_id).map(|fpga| (fpga, port_index))
-}
+// Notes:
+//
+// Address 1 FPGA at a time in each message, a const-sized bitmask (u32? u64?)
```

Comment:
`u16`?

Reply:
Lol, this should just go away those were my own notes.

Reply:
Removed in `c8233ff`